src/dex-ui/pages/CreateProposal/ProposalForms/ContractUpgradeProposalForm.tsx

```diff
@@ -1,16 +1,123 @@
-import { FormControl, Input } from "@chakra-ui/react";
+import { FormControl, Input, FormErrorMessage, Flex, Button, Spacer, Text } from "@chakra-ui/react";
+import { ReactElement, useState } from "react";
+import { useForm } from "react-hook-form";
+import { useNavigate } from "react-router-dom";
+import { TextEditor } from "../../../../dex-ui-components";
+import { useDexContext } from "../../../hooks";
+import { ProposalType } from "../../../store/governanceSlice";
+import { FileUploader } from "../../../../dex-ui-components";
+
+interface ContractUpgradeProposalFormData {
+  title: string;
+  description: string;
+  linkToDiscussion: string;
+  abiFile: string;
+  abiFileName: string;
+  newImplementationAddress: string;
+  proxyAddress: string;
+}
+
+export function ContractUpgradeProposalForm(): ReactElement {
+  const { governance } = useDexContext(({ governance }) => ({ governance }));
+  const navigate = useNavigate();
+  const {
+    handleSubmit,
+    register,
+    formState: { errors, isSubmitting },
+  } = useForm<ContractUpgradeProposalFormData>();
+
+  const handleCancelClick = () => navigate("/governance");
+
+  const onFileReadAction = (file: string, fileName: string) => {
+    console.log("File", file);
+    setAddress(fileName);
+  };
+  async function onSubmit(data: ContractUpgradeProposalFormData) {
+    // TODO: Call the Contract
```

Comment:
The `TokenTransferProposalForm` is still using the global Zustand store action to call `createProposal`. We can use this opportunity to move and wrap the `createProposal` function in a `react-query` hook (useCreateProposal) to improve the way we're submitting new proposals in both of these forms. Take a look at the `react-query` hooks under `hooks/governance` in the latest `main` code for examples. I can explain on a call in more detail if needed.

Reply:
Done